### PR TITLE
feat: better reset-password-form

### LINKED ISSUE #3876: Missing password validation in forgot password window
**Describe the bug**

In forgot password window , validation is missing while setting new password. Also it lacks toggle icon.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to 'forgot password'
2. Click on 'link sent to your mail'
3. Type invalid password 

**Expected behaviour**

Have validations similar to sign up and login.

**Screenshots**

Screenshot 2020-01-27 19:09:10

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Ubuntu 18.04
 - Browser :  chrome
 - Version : 79.0.3945.88 (Official Build) (64-bit)

**Additional context**

Working on it.

### PR DESCRIPTION
**Pull request by snitin315**
Fixes #3876 

#### Changes proposed in this pull request:

Screenshot from 2020-02-28 15-15-10

Screenshot from 2020-02-28 15-14-09

### CODE REVIEW

File: app/templates/components/forms/reset-password-form.hbs [hunk 1]
```diff
@@ -19,8 +19,24 @@
...
-                {{input type='password' name='password' placeholder=(t 'password') value=password}}
+                <i class="lock icon"></i>
+                  {{input type=(if showNewPass 'text' 'password') name='password' placeholder=(t 'Password') value=password}}
+                <div class="ui small basic icon buttons">
+                  <button type="button" class="ui button" id="eye2" {{ action 'showNewPassword'}}>
+                    <i class="{{if showNewPass 'hide' 'unhide'}} basic icon"></i>
+                  </button>
+                </div>
+              </div>
+              <br>
```
Comment (kushthedude) [inline app/templates/components/forms/reset-password-form.hbs]: Use semantic class

File: app/templates/components/forms/reset-password-form.hbs [hunk 2]
```diff
@@ -19,8 +19,24 @@
...
+              <br>
```
Comment (kushthedude) [inline app/templates/components/forms/reset-password-form.hbs]: Use semantic class

File: app/templates/components/forms/reset-password-form.hbs [hunk 3]
```diff
@@ -19,8 +19,24 @@
           <div class="field">
             {{#if token}}
               <div class="ui left icon input">
-                <i class="key icon"></i>
-                {{input type='password' name='password' placeholder=(t 'password') value=password}}
+                <i class="lock icon"></i>
+                  {{input type=(if showNewPass 'text' 'password') name='password' placeholder=(t 'Password') value=password}}
+                <div class="ui small basic icon buttons">
+                  <button type="button" class="ui button" id="eye2" {{ action 'showNewPassword'}}>
```
Comment (kushthedude) [inline app/templates/components/forms/reset-password-form.hbs]: No need for `type=button`

File: app/templates/components/forms/reset-password-form.hbs [hunk 4]
```diff
@@ -19,8 +19,24 @@
...
+                  </button>
+                </div>
+              </div>
+              <br>
+              <br>
+              <div class="ui left icon input">
+                <i class="lock icon"></i>
+                  {{input type=(if showConfirmPass 'text' 'password') name='password_repeat' placeholder=(t 'Confirm Password') value=password_repeat}}
+                <div class="ui small basic icon buttons">
+                  <button type="button" class="ui button" id="eye2" {{ action 'showConfirmPassword'}}>
```
Comment (kushthedude) [inline app/templates/components/forms/reset-password-form.hbs]: See above

File: app/templates/components/forms/reset-password-form.hbs [hunk 5]
```diff
@@ -19,8 +19,24 @@
...
+                  <button type="button" class="ui button" id="eye2" {{ action 'showNewPassword'}}>
+                    <i class="{{if showNewPass 'hide' 'unhide'}} basic icon"></i>
+                  </button>
+                </div>
+              </div>
+              <br>
+              <br>
+              <div class="ui left icon input">
+                <i class="lock icon"></i>
+                  {{input type=(if showConfirmPass 'text' 'password') name='password_repeat' placeholder=(t 'Confirm Password') value=password_repeat}}
```
Comment (kushthedude) [inline app/templates/components/forms/reset-password-form.hbs]: There is no need for `value=password_repeat`, we don't have an actual field hence the identifier name will do for same.

**Comment by snitin315** [comment]
@kushthedude Done :100:

**Pull request merged**